### ISSUE TITLE
Updating the Providers workflow

### ISSUE THREAD

**Issue by techtolentino** [Qiskit/qiskit.org#3049]
## Description
As we look ahead, the team anticipates that the Providers page will see a lot of updates that include: new categories are created, categories are updated, or removed... providers are added, providers get reordered, removed, etc... 

Given this scenario, we'll need to see how we can improve our workflow, to get as _close to_ automated as possible.

### Current workflow
In today's Provider page setup, this is our workflow:

- @HuangJunye updates content in `yml` files
- those yaml files are used with an automated script, and they produce a json file _per category_
- with those files, a new PR can be made into the qiskit org repo
- after the PR is checked to ensure everything renders correctly, the PR can be merged and the page is updated

#### Manual work
Also in today's workflow, if there is a new Provider category, there will be some manual dev work
- pull in new json in `providersContent.ts`
- add new category variables
- add new category to exported content sections + table of contents

### Future scenario
Imagine a scenario where this manual part can be _more_ automated

- `yml` files are updated
- script is ran, and produces new JSON
- PR is made
- the template is able to parse the JSON data to
  - create new sections
  - update providers lists 
  - render properly, with any new order
  
  
#### Summary
How can we improve our current workflow so that we can minimize the manual dev effort, and improve delivery of updated providers content?

### Additional context

cc: @1ucian0 @HuangJunye

**Comment by y4izus** [comment]
Update on the workflow:

- Now we have the content directly on YAML files, so we have removed the 2 first steps of the process (create the file on JSON and convert it to YAML)

In order to make the workflow more automatic, we need to know where the providers info lives, do we have that info? @HuangJunye @1ucian0 @javabster . Also, I think this issue exceeds what we pretend with the providers page, so I think it should go to a new milestone "Providers 2.0", do you agree?

**Comment by HuangJunye** [comment]
I think we can directly modify the YAML files and the information lives in the YAML files. It should be straightforward to update existing provider information by directly modifying the YAML files. For new providers, we can either manually write the YAML files or use the scripts (need to be updated to output YAML files instead of JSON) in https://github.com/HuangJunye/qiskit-providers-quickstart to generate the resulting YAML files. 

Now that qiskit.org is consuming YAML files directly, the main difference between the input YAML files in https://github.com/HuangJunye/qiskit-providers-quickstart/tree/main/yaml_in and the output is in the code examples. The input YAML files contain only provider specific imports and the script takes that to combine with code examples in [algorithms.yaml](https://github.com/HuangJunye/qiskit-providers-quickstart/blob/main/yaml_in/algorithms.yaml) to generate multiple code examples. I don't foresee new providers will be added very frequently, at most once every few months. Therefore we don't need to automate this process. I will handle the update.

In summary, the update workflow is simplified to only one step:

1. Modify providers YAML files.

**Comment by HuangJunye** [comment]
Related to this issue, we should add tests to check the code examples are correct. I remember @y4izus mentioned that you don't want python files to live in the repo. Can we generate python files (from YAML files code examples field) in a github workflow on the fly and tests them? That way, no python files will live in the repo. 

I can create a separate issue for that and add that to "Providers 2.0" milestone.

**Comment by y4izus** [comment]
Per offline conversations:

- We can close this issue
- It wasn't about having python files on the repo, it was that I didn't want to have non-related code on the repo
- If the tests for that code are done in a different repo, please, add the issue there. We would like to have the repo as clean as possible :)
